php_phongo.c

```diff
@@ -953,6 +952,13 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t *uri, bson_t *options T
 			continue;
 		}
 
+        if (!strcasecmp(key, MONGOC_URI_READPREFERENCE)) {
```

Comment:
You are right, also in this case we may make use of the `bson_iter_key(&iter);` by using it with the Unsupported readPreference value ... as follows:

```php
if (bson_iter_init_find_case(&iter, options, MONGOC_URI_READPREFERENCE)) {
	const char *key = bson_iter_key(&iter);
	if (BSON_ITER_HOLDS_UTF8(&iter)) {
		const char *str = bson_iter_utf8(&iter, NULL);
		/* ... */
		} else {
		    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported %s value: '%s'", key, str);
		}
	} else {
		/* phongo_throw_exception() for invalid type */
		return false;
	}
}
```